core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/customize-interaction-modal.component.ts

```diff
@@ -0,0 +1,465 @@
+import { AppConstants } from 'app.constants';
+import { StateEditorService } from 'components/state-editor/state-editor-properties-services/state-editor.service';
+import { ContinueValidationService } from 'interactions/Continue/directives/continue-validation.service';
+import { EndExplorationValidationService } from 'interactions/EndExploration/directives/end-exploration-validation.service';
+import { AlgebraicExpressionInputValidationService } from 'interactions/AlgebraicExpressionInput/directives/algebraic-expression-input-validation.service';
+import { ImageClickInputValidationService } from 'interactions/ImageClickInput/directives/image-click-input-validation.service';
+import { ItemSelectionInputValidationService } from 'interactions/ItemSelectionInput/directives/item-selection-input-validation.service';
+import { NumberWithUnitsValidationService } from 'interactions/NumberWithUnits/directives/number-with-units-validation.service';
+import { NumericExpressionInputValidationService } from 'interactions/NumericExpressionInput/directives/numeric-expression-input-validation.service';
+import { NumericInputValidationService } from 'interactions/NumericInput/directives/numeric-input-validation.service';
+import { DragAndDropSortInputValidationService } from 'interactions/DragAndDropSortInput/directives/drag-and-drop-sort-input-validation.service';
+import { GraphInputValidationService } from 'interactions/GraphInput/directives/graph-input-validation.service';
+import { SetInputValidationService } from 'interactions/SetInput/directives/set-input-validation.service';
+import { CodeReplValidationService } from 'interactions/CodeRepl/directives/code-repl-validation.service';
+import { MathEquationInputValidationService } from 'interactions/MathEquationInput/directives/math-equation-input-validation.service';
+import { MultipleChoiceInputValidationService } from 'interactions/MultipleChoiceInput/directives/multiple-choice-input-validation.service';
+import { PencilCodeEditorValidationService } from 'interactions/PencilCodeEditor/directives/pencil-code-editor-validation.service';
+import { TextInputValidationService } from 'interactions/TextInput/directives/text-input-validation.service';
+import { InteractiveMapValidationService } from 'interactions/InteractiveMap/directives/interactive-map-validation.service';
+import { MusicNotesInputValidationService } from 'interactions/MusicNotesInput/directives/music-notes-input-validation.service';
+import { FractionInputValidationService } from 'interactions/FractionInput/directives/fraction-input-validation.service';
+import { RatioExpressionInputValidationService } from 'interactions/RatioExpressionInput/directives/ratio-expression-input-validation.service';
+import { Warning } from 'interactions/base-interaction-validation.service';
+import cloneDeep from 'lodash/cloneDeep';
+
+interface CustomizationArgSpecsInterface {
+  name: string;
+  value: unknown;
+  default_value: unknown;
```

Comment:
why unknown type is used here?

Reply:
default_value has too many types and those have more subtypes